perfkitbenchmarker/openstack/os_virtual_machine.py

```diff
@@ -0,0 +1,154 @@
+import logging
+import random
+import string
+import time
+
+from perfkitbenchmarker import virtual_machine, linux_virtual_machine
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.openstack import utils as os_utils, os_disk
+
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('os_auth_url', 'http://localhost:5000',
+                    'This determine url to Keystone authenticate service.'
+                    'It is require to discovery other OpenStack services URLs')
+
+flags.DEFINE_string('os_username', 'admin',
+                    'OpenStack login')
+
+flags.DEFINE_string('os_passwd', 'admin',
+                    'OpenStack password')
+
+flags.DEFINE_string('os_tenant', 'admin',
+                    'OpenStack tenant name')
+
+flags.DEFINE_boolean('os_config_drive', False,
+                     'Add possibilities to get metadata from external drive')
+
+
+def id_generator(size=6, chars=string.ascii_uppercase + string.digits):
+    return ''.join(random.choice(chars) for _ in range(size))
+
+
+class OpenStackVirtualMachine(virtual_machine.BaseVirtualMachine):
+    """Object representing an OpenStack Virtual Machine"""
+
+    instance_counter = 0
+
+    def __init__(self, vm_spec):
+        super(OpenStackVirtualMachine, self).__init__(vm_spec)
+        self.special_id = id_generator()
+        self.name = 'perfkit_vm_%d_%s' % (self.instance_counter, self.special_id)
+        self.key_name = 'perfkit_key_%d_%s' % (self.instance_counter, self.special_id)
+        OpenStackVirtualMachine.instance_counter += 1
```

Comment:
We've generally been deferring name to the superclass. There's a global instance counter (protected by a lock) defined [here](https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/blob/master/perfkitbenchmarker/virtual_machine.py#L101).

`self.name` will already be defined, since you've called the superclass constructor.